### PR TITLE
Plan model

### PR DESCRIPTION
**Pull request by RDiMag**

### CODE REVIEW

**Review by theodorusandi — APPROVED** [review]
nice!

**Review by Marcellia — APPROVED** [review]

File: src/app/models/PlanModel.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import * as mongodb from 'mongodb';
+
+class PlanModel {
+  _id: mongodb.ObjectId;
+  name: string;
+  description: string;
+  thumbnail: string;
+  tags: string[];
+  likes: number;
+  classes: string[];
```
Comment (jpatrici) [inline src/app/models/PlanModel.ts:10]:
```suggestion
  classes: ClassInterface[];
```

File: src/app/models/PlanModel.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import * as mongodb from 'mongodb';
```
Comment (jpatrici) [inline src/app/models/PlanModel.ts:1]:
```suggestion
import { ClassInterface } from './ClassModel';
const { defaultProfilePicture } = require('../../config/defaultVars.config');
```

File: src/app/models/PlanModel.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  likes: number;
+  classes: string[];
+  createdBy: string;
+  visible: boolean;
+  createdAt = Date;
+
+  constructor(
+    name = '',
+    description = '',
+    thumbnail = '',
```
Comment (jpatrici) [inline src/app/models/PlanModel.ts:18]:
```suggestion
    thumbnail = defaultProfilePicture,
```

File: src/app/models/PlanModel.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import * as mongodb from 'mongodb';
+
+class PlanModel {
+  _id: mongodb.ObjectId;
+  name: string;
+  description: string;
+  thumbnail: string;
```
Comment (jpatrici) [inline src/app/models/PlanModel.ts:7]:
```suggestion
  thumbnail: File;
```

File: src/app/models/PlanModel.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  constructor(
+    name = '',
+    description = '',
+    thumbnail = '',
+    tags = [],
+    _likes = 0,
+    classes = [],
+    createdBy = '',
+    visible = false,
+    createdAt = Date,
```
Comment (jpatrici) [inline src/app/models/PlanModel.ts:24]:
```suggestion
    createdAt = new Date(),
```

File: src/app/models/PlanModel.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  createdBy: string;
+  visible: boolean;
+  createdAt = Date;
+
+  constructor(
+    name = '',
+    description = '',
+    thumbnail = '',
+    tags = [],
+    _likes = 0,
```
Comment (jpatrici) [inline src/app/models/PlanModel.ts:20]:
```suggestion
    likes = 0,
```

File: src/app/models/PlanModel.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    this.tags = tags;
+    this.likes = 0;
+    this.classes = classes;
+    this.createdBy = createdBy;
+    this.visible = visible;
+    this.createdAt = createdAt;    
+  }
+}
+
+export default PlanModel;
```
Comment (jpatrici) [inline src/app/models/PlanModel.ts:40]: add a new line at end of file please 👍

File: src/app/models/PlanModel.ts (hunk 2 above)
Comment (RDiMag) [reply src/app/models/PlanModel.ts:1]: @jpatrici When I removed the mongodb import, I am now getting TS errors "cannot find namespace or name 'mongodb'". This affects lines 5 and 28. Is this an actual problem?

File: src/app/models/PlanModel.ts (hunk 5 above)
Comment (RDiMag) [reply src/app/models/PlanModel.ts:24]: @jpatrici I made this change, but I am getting this error: "Type 'Date' is missing the following properties from type 'DateConstructor': prototype, parse, UTC, now"

Comment (jpatrici) [reply src/app/models/PlanModel.ts:24]: but this is not the way to create a Date. What you need to do here is giving a default value to `createdAt`, and it should be `new Date()`, `Date` is the type. I tried it locally and I have no problem and didnt get any error

File: src/app/models/PlanModel.ts (hunk 2 above)
Comment (jpatrici) [reply src/app/models/PlanModel.ts:1]: I made a mistake in commenting, don't remove the mongodb import, was meant to just add the class interface and the defaultProfilePicture

Comment (RDiMag) [reply src/app/models/PlanModel.ts:1]: Okay! I will change it. I bet this fixes the date issue as well.

File: src/app/models/PlanModel.ts (hunk 5 above)
Comment (RDiMag) [reply src/app/models/PlanModel.ts:24]: I had a syntax error earlier in the file. Fixed!

**Pull request merged**